src/invidious/yt_backend/youtube_api.cr

```diff
@@ -666,6 +670,52 @@ module YoutubeAPI
     return initial_data
   end
 
+  ####################################################################
+  # _post_invidious_companion(endpoint, data)
+  #
+  # Internal function that does the actual request to Invidious companion
+  # and handles errors.
+  #
+  # The requested data is an endpoint (URL without the domain part)
+  # and the data as a Hash object.
+  #
+  def _post_invidious_companion(
+    endpoint : String,
+    data : Hash
+  ) : Hash(String, JSON::Any)
+    headers = HTTP::Headers{
+      "Content-Type"  => "application/json; charset=UTF-8",
+      "Authorization" => "Bearer " + CONFIG.invidious_companion_key,
+    }
+
+    # Logging
+    LOGGER.debug("Invidious companion: Using endpoint: \"#{endpoint}\"")
+    LOGGER.trace("Invidious companion: POST data: #{data}")
+
+    # Send the POST request
+
+    begin
+      response = make_client(CONFIG.invidious_companion.sample,
+        &.post(endpoint, headers: headers, body: data.to_json))
+      body = response.body
+      if (response.status_code != 200)
+        raise Exception.new("Error while communicating with Invidious companion: \
+                              status code: " + response.status_code.to_s + " and body: " + body)
+      end
+    rescue ex
+      raise InfoException.new("Error while communicating with Invidious companion: " + (ex.message || "no extra info found"))
+    end
+
+    if body.nil?
+      raise InfoException.new("Error while communicating with Invidious companion: no response data.")
+    end
```

Comment:
This is not required, as `body` is always `String`, because
1. You're using [`response.body`](https://crystal-lang.org/api/1.13.1/HTTP/Client/Response.html#body%3AString-instance-method) and not [`response.body?`](https://crystal-lang.org/api/1.13.1/HTTP/Client/Response.html#body%3F%3AString%7CNil-instance-method)
2. You're already catching any potential errors above!